oxenss/server/omq.cpp

```diff
@@ -56,6 +61,40 @@ void OMQ::handle_sn_data_ready(oxenmq::Message& message) {
     if (!service_node_->is_swarm_peer(xpk))
         return message.send_reply("Swarm mismatch");
 
+    std::optional<oxenss::snode::contact> ct = service_node_->contacts().find(xpk);
+    if (!ct)
+        return message.send_reply("Contact info missing");
+
+    if (ct->version >= snode::SN_DATA_READY_WITH_REQUEST_VERSION) {
+        if (message.data.empty())
+            return message.send_reply("Request payload missing");
+
+        snode::SerialiseDataReadyRequestResult deserialised =
+                snode::serialise_data_ready_request(Serialise::Read, message.data[0], {});
+        if (!deserialised.bt.success)
+            return message.send_reply("Request payload malformed {}"_format(deserialised.bt.error));
+
+        const snode::DataReadyRequest& request = deserialised.request;
+        if (request.needs_db_dump)
+            service_node_->set_member_needs_db_dump(crypto::legacy_pubkey{ct->pubkey_ed25519});
+
+        if (auto level = log::Level::debug; log::get_level(logcat) <= level) {
+            std::string label;
+            if (deserialised.bt.success)
+                label = "rejected, bad request payload. {})"_format(deserialised.bt.error);
+            else
+                label = "rejected due to bad request args";
+
+            log::log(
+                    logcat,
+                    level,
```

Comment:
`level` here is invariant, and is only used for the initial comparison, and in `log::log` here instead of `log::debug`; this does not improve clarity.  The whole thing can be simplified (both computationally and in terms of readability) as:
```suggestion
        if (log::get_level(logcat) <= log::Level::debug) {
            std::string label;
            if (deserialised.bt.success)
                label = "rejected, bad request payload. {})"_format(deserialised.bt.error);
            else
                label = "rejected due to bad request args";

            log::debug(
                    logcat,
```